gtsfm/loader/colmap_loader.py

```diff
@@ -0,0 +1,183 @@
+"""Simple loader class that reads a dataset with metadata formatted in the COLMAP style.
+
+Authors: John Lambert
+"""
+
+import os
+from pathlib import Path
+from typing import Optional
+
+from gtsam import Cal3Bundler, Pose3
+
+import gtsfm.utils.images as img_utils
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+logger = logger_utils.get_logger()
+
+
+class ColmapLoader(LoaderBase):
+    """Simple loader class that reads a dataset with metadata formatted in the COLMAP style.
+
+    Note: assumes all images are of the same dimensions.
+    TODO: support max-resolution constraint for images are of the same dimensions.
+       Would introduce a constraint between get_image() and get_intrinsics()
+
+    Folder layout structure:
+    - RGB Images: images/
+    - Intrinsics data (optional): cameras.txt (see https://colmap.github.io/format.html#cameras-txt)
+    - Extrinsics data (optional): images.txt (see https://colmap.github.io/format.html#images-txt)
+
+    If explicit intrinsics are not provided, the exif data will be used.
+    """
+
+    def __init__(
+        self,
+        colmap_files_dirpath: str,
+        images_dir: str,
+        use_gt_intrinsics: bool = True,
+        use_gt_extrinsics: bool = True,
+        max_frame_lookahead: int = 1,
+        max_resolution: int = 760,
+    ) -> None:
+        """Initializes to load from a specified folder on disk.
+
+        Args:
+            colmap_files_dirpath: path to directory containing COLMAP-exported data, with images.txt
+                and cameras.txt files
+            images_dir: path to directory containing images files
+            use_gt_intrinsics: whether to use ground truth intrinsics. If COLMAP calibration is
+               not found on disk, then use_gt_intrinsics will be set to false automatically.
+            use_gt_extrinsics: whether to use ground truth extrinsics
+            max_frame_lookahead: if images were sequentially captured, maximum number
+               of consecutive frames to consider for matching/co-visibility. Defaults to 1, i.e.
```

Comment:
1 doesn't imply data was sequentially captured. In fact, any value of max_frame_lookahead assumes data is sequentially captured.

Reply:
How about "1 implies data was sequentially captured by a fast moving camera with a low frame rate"?

Reply:
I think we do not need to add any comment. Your statement is true in general, but there might be other reasons why we use the value 1.

Reply:
is this better?
```
            max_frame_lookahead: maximum number of consecutive frames to consider for
                matching/co-visibility. Any value of max_frame_lookahead less than the size of
                the dataset assumes data is sequentially captured
```